build.gradle

```diff
@@ -22,21 +42,51 @@ plugins {
 }
 
 
+apply from: 'gradle/formatting.gradle'
+apply plugin: 'maven-publish'
+apply plugin: 'java'
+apply plugin: 'opensearch.opensearchplugin'
+apply plugin: 'opensearch.testclusters'
+
+opensearchplugin {
```

Comment:
Can you create an issue on creating a new gradle plugin `opensearchextension` which shouldn't relay on plugin.descriptor property file? That's the reason for `./gradlew run` not working because of these changes.

Reply:
But how would that work when we need to run plugins and extensions at the same time?

Reply:
SDK would inherit `opensearchextension` gradle plugin and we need to figure out a way to run it either by creating a new file like the descriptor property or by completely removing it.

Reply:
So you're saying we would have to inherit both an extensions plugin and the current plugin for gradle?

Reply:
In that scenario, `opensearchextension` would have to contain all the current code to run plugins, except for the plugin-descriptor part, and contain the additions I made to run extensions

Reply:
Plugins will still keep using `opensearchplugin`.

Reply:
`opensearchextension` should consider all these use cases.

Reply:
I'll create an issue for the gradle run failure that we're seeing, but we should have further discussion about the solution.

Reply:
https://github.com/opensearch-project/opensearch-sdk-java/issues/789